File: Recursion_and_Backtracking_pepcoding_level_1/2.Recursion_on_array/3.max_in_array.js
const max_in_arr = (arr, i = 0) => {
    if (i === arr.length - 1) return arr[i];

    const child = max_in_arr(arr, i + 1);
    return Math.max(arr[i], child);
};

console.log(max_in_arr([-5, 34, 342, 34234, 32]));

// E
// max of( 0 to n)

// F
// max of( 1 to n)

// F + E
// max of (0 , 1 to n)

File: Recursion_and_Backtracking_pepcoding_level_1/2.Recursion_on_array/3.max_in_array.rb
def max_in_arr arr, i=0
  return -Float::INFINITY if i == arr.size

  maxi = max_in_arr arr, i+1
  return [arr[i], maxi].max
end

puts max_in_arr [-5, 34, 342, 34234, 32]
puts max_in_arr [34234, 32,-5, 34, 342]
